audinterface/core/feature.py

```diff
@@ -468,58 +557,27 @@ def _values_to_frame(
             win_dur = None
             hop_dur = None
 
-        if self.process.process_func_is_mono and self.num_channels > 1:
-            features = np.concatenate(features)
-
-        if not isinstance(features, np.ndarray):
-            raise RuntimeError(
-                "Features must be a 'np.ndarray', "
-                f"not '{type(features)}'."
-            )
-
-        # features = np.array(features)
-        if features.ndim < 2 or features.ndim > 3:
-            raise RuntimeError(
-                f'Dimension of extracted features must be 2 or 3, '
-                f'not {features.ndim}.'
-            )
-
-        # Force third time step dimension
-        features = np.atleast_3d(features)
-        n_channels = features.shape[0]
-        n_features = features.shape[1]
-        n_time_steps = features.shape[2]
-
-        if n_channels != self.num_channels:
-            raise RuntimeError(
-                f'Number of channels must be {self.num_channels}, '
-                f'not {n_channels}.'
-            )
-        if n_features != len(self.feature_names):
-            raise RuntimeError(
-                f'Number of features must be {len(self.feature_names)}, '
-                f'not {n_features}.'
-            )
+        features = self._assert_shape_3d(features)
```

Comment:
Maybe not call it `_assert_shape_3d` as it will not return `True` or `False` but will change the features.

Reply:
Renamed to `_reshape_3d()`.